azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultPropertySource.java

```diff
@@ -8,9 +8,9 @@
 
 import static com.microsoft.azure.keyvault.spring.Constants.AZURE_KEYVAULT_PROPERTYSOURCE_NAME;
 
-import org.springframework.core.env.PropertySource;
+import org.springframework.core.env.EnumerablePropertySource;
 
-public class KeyVaultPropertySource extends PropertySource<KeyVaultOperation> {
+public class KeyVaultPropertySource extends EnumerablePropertySource<KeyVaultOperation> {
```

Comment:
May I ask why we change this back to EnumerablePropertySource here? Since @chenrujun changed this in #882.

Reply:
For it to show up in the /env actuator endpoint it needs to be an EnumerablePropertySource.

Reply:
@mnriem Thank you for your work. 

For `EnumerablePropertySource` , We have a case like this: When we have many properties, for example, 150 properties. If we use EnumerablePropertySource, it will cost a lot of time(few minutes, maybe longer) to read all of these properties. And some spring framework logic will get all propertValues more than one time. Here is an example: [spring-config get all propertValues](https://github.com/spring-cloud/spring-cloud-commons/blob/master/spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/encrypt/EnvironmentDecryptApplicationInitializer.java#L230-L235)


Could you please cache `propertyVaules` , just like cache `propertyNames`.

1. Use same config item to set the expire time.
2. Set default expire time to 1 minute.

Reply:
I am in the process of addressing performance issues in PR #905, in which indeed the keys and values are cached.